packages/web3-signature/test/web3-signature-provider.test.ts

```diff
@@ -16,34 +18,46 @@ const id1Raw = {
 };
 
 const data = { What: 'ever', the: 'data', are: true };
-const normalizedData = Utils.crypto.normalize(data);
+// const normalizedData = Utils.crypto.normalize(data);
+const hashData = Utils.crypto.normalizeKeccak256Hash(data).value;
+const signatureValueExpected = Utils.crypto.EcUtils.sign(id1Raw.signatureParams.privateKey, hashData);
 
-const mockEth: any = {
-  personal: {
-    async sign(): Promise<any> {
-      return;
-    },
-  },
-};
+const mockWeb3: any = {
+  getSigner: jest.fn().mockImplementation(() => ({signMessage: () => {return signatureValueExpected;} }))
+}
 
 /* tslint:disable:no-unused-expression */
 describe('web3-signature-provider', () => {
   describe('sign', () => {
     it('can sign', async () => {
-      const spy = jest.spyOn(mockEth.personal, 'sign');
-      const signProvider = new Web3SignatureProvider('http://localhost:8545');
+      const signProvider = new Web3SignatureProvider(new providers.InfuraProvider());
```

Comment:
If Infura is only used for the object creation but not actually used, you should precise that in a comment so we don't incriminate this external service if, for example, unit tests fail some day.